### ISSUE TITLE
✋🏽Design Decisions for Hand-Labeled Features

### ISSUE THREAD

**Issue by xehu** [Watts-Lab/team_comm_tools#49]
# Background / Recap
We want to know: **which types of team communication processes matter in different team activities?** Our goal is to extract theories of teams' communication processes (e.g., "teams perform better when members speak more positively than negatively") and "map"  these theories to the different contexts (tasks) in which teams operate. For example, does such positive thinking matter less when the activity is analytical, rather than creative?

The crux of our work now is in extracting features that match these theories of team processes, as closely as possible. Here is where we encounter a design decision: specifically, what should the role of hand-labeling be in this process?

# Overview of the Problem
Many of the most interesting features in a team's communication process are complex and difficult to quantify --- for example, a member's "assertiveness," or a team's level of "coordination" and "cooperation," certainly leave traces in what team members say, but can elude precise definition. Therefore, it may be important to consider the role that hand-labeling features can play in helping to quantify these difficult-to-measure behavioral features.

There are three potential uses for hand-labeling:
1. **As features.** This is essentially the approach taken by Hancheng and Mark, et al. in their 2020 CSCW paper [(Cao et al. 2020)](https://dl.acm.org/doi/pdf/10.1145/3432929?casa_token=B6hmir1dHdcAAAAA:I6rRD0bR0EvACZ4CpWK9tKUQFgPXtNTKT7BD-TSPiHu0mYXAwNflHot94ky1pwAegrDSaDhpOuizDA). Here, they asked humans to rate 20 conversational features sourced from the organizational behavior literature (e.g., "generating many ideas," "positive interaction," "passive-aggressive interaction").  The advantage of obtaining these labels is the ability to potentially augment predictive models with human insight. However, they found that these hand-labeled features were ultimately not very predictive of whether a team chooses to work together again after an interaction (their dependent variable of interest), so it is questionable whether this approach is worth the requisite cost.
2. **As training data.** Alternatively, we can computationally define all features, using human labels in training process. In training, human labels can either be used for supervised learning (e.g., see paper [C.9](https://web.stanford.edu/~jurafsky/pubs/ranganath2013.pdf)), or as part of the input in a specific formula (e.g., paper [C.12](https://aclanthology.org/C16-1181.pdf)); both of these examples are discussed in more detail later. To do this, we would likely need to split off a portion of our data for training purposes; but since we only have a few hundred teams, this could waste a precious resource. An alternative is to pre-train computational feature extractors on some other dataset (e.g., Twitter or Yelp reviews), and then fine-tune using a smaller subset of our real data.
3. **As validation data.**  But if we go with option (2), how would we know if our computational features are accurate proxies of these behavioral features? We probably need to also pre-label a "validation set" as well. But if we're pre-labeling all of this --- and if the human judgment is ultimately the gold standard that these fancy models are meant to approximate --- we then end up back at (1). Effectively, if we're labeling that much, shouldn't we just use hand-labeling for everything?

# What (and how much) should we hand-label?
We reach a decision point with a few considerations:
- Hand-labels can be a great "gold standard," as human judgments offer a nuanced perspective that can either augment models or help to benchmark them.
- However, hand-labeling everything can be costly.
- Additionally, Hancheng and Mark's work shows that hand labels may not add much value.
- We can include hand labels in the training and validation process.
- But if use hand labels in training/validation, this (1) still becomes costly; (2) causes us to lose a chunk of our dataset due to the train/test split; and (3) assumes that hand labels are the "ground truth" (which may or may not be the right assumption).

# Appendix: Learning by (Previous Authors') Examples
Here are two examples of previous work that uses human labeling in different ways:

- [C.9](https://web.stanford.edu/~jurafsky/pubs/ranganath2013.pdf) trains a classifier to predict whether a conversation is "friendly," "flirtatious," "awkward," or "assertive." The input features are the prosodic and lexical features of the conversation, and human labels (done by the participants of the conversation) are used for supervised learning.

- [C.12](https://aclanthology.org/C16-1181.pdf) uses humans to annotate political interviews for turn segmentation and content (e.g., is the answer informative/responsive? Does the speaker seem satisfied with the response?). They then use these annotations to calculate the extent to which a speaker is "cooperative" or not: the "degree of cooperation" is defined as the proportion of a speaker's features that are rated as cooperative, divided by the sum of all cooperative and non-cooperative features.

## Other Ideas
- Could we use the weights from the pre-trained model in [C.9](https://web.stanford.edu/~jurafsky/pubs/ranganath2013.pdf) to predict constructs like "assertiveness?" (But then, we'd still need validation data...)

**Comment by linneagandhi** [comment]
Maybe Alison Woods Brooks would have an idea, since she studies conversations & has been doing work with Susan Athey? It's possible they have some computational (non manual) way of coding tone or other similar attributes

**Comment by xehu** [comment]
@linneagandhi @JamesPHoughton just to follow up on this -- I'd love to make the connection to connect with Allison and Susan!

**Comment by xehu** [comment]
Bumping this issue following 2nd year paper presentation --- seems like this might be a priority as people will ask us about validation.

**Comment by xehu** [comment]
Closing as this is about a previous version of the project and no longer related to the toolkit
